Partially implement `_CFXPCCreateCFObjectFromXPCObject`

facekapow opened the pull request:
Title says it all.

facekapow committed:
Partially implement `_CFXPCCreateCFObjectFromXPCObject`

bugaevc commented on CFXPCBridge.c:
```diff
@@ -29,14 +29,86 @@
+	if (type == XPC_TYPE_DICTIONARY) {
+		size_t count = xpc_dictionary_get_count(xo);
+		__block CFStringRef* keys = malloc(count * sizeof(CFStringRef));
+		__block CFTypeRef* values = malloc(count * sizeof(CFTypeRef));
```
I don't think the pointers have to be `__block`

bugaevc commented on CFXPCBridge.c:
```diff
@@ -29,14 +29,86 @@
+
+	if (type == XPC_TYPE_BOOL) {
+		CFBooleanRef boolean = xpc_bool_get_value(xo) == true ? kCFBooleanTrue : kCFBooleanFalse;
+		CFRetain(boolean); // are we supposed to do this? ¯\_(ツ)_/¯
```
Yes (but it should be a no-op).

bugaevc commented on CFXPCBridge.c:
```diff
@@ -29,14 +29,86 @@
+
+	if (type == XPC_TYPE_INT64) {
+		int64_t val = xpc_int64_get_value(xo);
+	}
```
Missing return

bugaevc commented on CFXPCBridge.c:
```diff
@@ -29,14 +29,86 @@
+
+	if (type == XPC_TYPE_NULL) {
+		CFRetain(kCFNull); // again, do we have to retain constants?
+		return kCFNull;
```
You can even write it in one line, like in objc: `return CFRetain(kCFNull);`

facekapow committed:
Remove unnecessary `__block`s; implement XPC int64 conversion

facekapow commented:
Removed unnecessary `__block`s, simplified `CFRetain` code, and implemented `int64` XPC -> CF conversion (`uint64` is almost exactly the same code, so when I implemented it I thought I had implemented `int64`).
